lib/decompress/zstd_decompress_block.c

```diff
@@ -776,7 +864,8 @@ size_t ZSTD_execSequenceEnd(BYTE* op,
     assert(oLitEnd < op + sequenceLength);
 
     /* copy literals */
-    ZSTD_safecopy(op, oend_w, *litPtr, sequence.litLength, ZSTD_no_overlap);
+    RETURN_ERROR_IF(op > *litPtr && op < *litPtr + sequence.litLength, dstSize_tooSmall, "output should not catch up to and overwrite literal buffer");
```

Comment:
Could it happen ?
Or should the code guarantee that this will never happen ?
aka: is it an error, or an `assert()` ?